src/components/ThemePropBlock.js

```diff
@@ -68,6 +76,14 @@ export default class ThemePropBlock extends React.Component {
         };
     }
 
+
+    valueUserInputHandler(event) {
+
```

Comment:
Yep! it's the right place to pass a new prop up to the root [themesOverrideList](https://github.com/sm-react/storybook-addon-material-ui/blob/751b0d2078bb541e2d24fc855b19d3e5c217c6b4/src/muiTheme.js#L42)
Check how we do it [here](https://github.com/sm-react/storybook-addon-material-ui/blob/751b0d2078bb541e2d24fc855b19d3e5c217c6b4/src/components/ThemePropBlock.js#L67):
```
this.props.onThemeTableOverride(propName, event.target.value);
```
`propName` - in your case will be setted by user
second arg - is a value of this prop

check [here](https://github.com/sm-react/storybook-addon-material-ui/blob/751b0d2078bb541e2d24fc855b19d3e5c217c6b4/src/components/ThemePropItem.js#L216) how we invoke this (note that the `valueHandler` is a *HightOder* function wich returns another function)
tip: I suppose the best way is to make an intermediate function somewere in `ThemePropItemInput.js` because you need to handle several events from all inputs, but it's up to you. If you'll find it too tricky I help you!